src/protagonist/CleanupHandler/AssetUpdatedHandler.cs

```diff
@@ -0,0 +1,357 @@
+﻿using System.IO.Enumeration;
+using System.Text.Json;
+using CleanupHandler.Infrastructure;
+using CleanupHandler.Repository;
+using DLCS.AWS.ElasticTranscoder;
+using DLCS.AWS.S3;
+using DLCS.AWS.S3.Models;
+using DLCS.AWS.SQS;
+using DLCS.Core.Collections;
+using DLCS.Core.Types;
+using DLCS.Model.Assets;
+using DLCS.Model.Assets.Metadata;
+using DLCS.Model.Messaging;
+using DLCS.Model.Policies;
+using DLCS.Repository.Messaging;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Version = IIIF.ImageApi.Version;
+
+namespace CleanupHandler;
+
+public class AssetUpdatedHandler  : IMessageHandler
```

Comment:
General observation on this class - is it worth having a 'transaction'. That could be done with any DB updates (ie only call `.SaveChanges()` once everything has been done). Similar approach to deleting keys/folders from S3 would be to run through and build up a list of all the operations to carry out and then actually do them at the end.

```cs
public async Task<bool> HandleMessage(QueueMessage message, CancellationToken cancellationToken = default)
{
  try
  {
    // as-is but don't actually call S3, just build up lists of keys to delete
    // means that stuff isn't left in a half removed state
    await CleanupRemoved();
    await CleanupChanges();
    
    // now do the actual removal / commit of transactino
    await dbContext.SaveChanges();
    await DeleteFromS3(keysToDelete);
  }
  catch (Exception) { return false ; }
}